frontend-public/src/tests/actionCreators/varianceActionCreator.spec.js

```diff
@@ -0,0 +1,162 @@
+import {
+  createVariance,
+  fetchVariancesByMine,
+  updateVariance,
+  addDocumentToVariance,
+  removeDocumentFromVariance,
+} from "@/actionCreators/varianceActionCreator";
+import * as genericActions from "@/actions/genericActions";
+import * as API from "@/constants/API";
+import * as MOCK from "@/tests/mocks/dataMocks";
+import { ENVIRONMENT } from "@/constants/environment";
+
+const dispatch = jest.fn();
+const requestSpy = jest.spyOn(genericActions, "request");
+const successSpy = jest.spyOn(genericActions, "success");
+const errorSpy = jest.spyOn(genericActions, "error");
+const mockAxios = new MockAdapter(axios);
+
+beforeEach(() => {
+  mockAxios.reset();
+  dispatch.mockClear();
+  requestSpy.mockClear();
+  successSpy.mockClear();
+  errorSpy.mockClear();
+});
+
+describe("`createVariance` action creator", () => {
+  const mineGuid = "1234567";
+  const mockPayload = {
+    effective_date: "1970-01-01",
+    expiry_date: "9999-12-31",
+  };
+  const url = ENVIRONMENT.apiUrl + API.VARIANCES(mineGuid);
+  it("Request successful, dispatches `success` with correct response", () => {
+    const mockResponse = { data: { success: true } };
+    mockAxios.onPost(url).reply(200, mockResponse);
+    return createVariance({ mineGuid }, mockPayload)(dispatch).then(() => {
+      expect(requestSpy).toHaveBeenCalledTimes(1);
+      expect(successSpy).toHaveBeenCalledTimes(1);
+      expect(dispatch).toHaveBeenCalledTimes(5);
+    });
+  });
+
+  it("Request failure, dispatches `error` with correct response", () => {
+    mockAxios.onPost(url).reply(400, MOCK.ERROR);
+    return createVariance({ mineGuid }, mockPayload)(dispatch).then(() => {
+      expect(requestSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(dispatch).toHaveBeenCalledTimes(4);
+    });
+  });
+});
+
+describe("`fetchVariancesByMine` action creator", () => {
+  const mineGuid = "1234567";
+  const url = ENVIRONMENT.apiUrl + API.VARIANCES(mineGuid);
+  it("Request successful, dispatches `success` with correct response", () => {
+    const mockResponse = { data: { success: true } };
+
+    mockAxios.onGet(url).reply(200, mockResponse);
+    return fetchVariancesByMine(mineGuid)(dispatch).then(() => {
+      expect(requestSpy).toHaveBeenCalledTimes(1);
+      expect(successSpy).toHaveBeenCalledTimes(1);
+      expect(dispatch).toHaveBeenCalledTimes(6);
+    });
+  });
+
+  it("Request failure, dispatches `error` with correct response", () => {
+    mockAxios.onGet(url, MOCK.createMockHeader()).reply(400, MOCK.ERROR);
+    return fetchVariancesByMine(mineGuid)(dispatch).then(() => {
+      expect(requestSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(dispatch).toHaveBeenCalledTimes(4);
+    });
+  });
+});
+
+describe("`updateVariance` action creator", () => {
+  const mineGuid = "1234567";
+  const mockPayload = {
+    effective_date: "1970-01-01",
+    expiry_date: "9999-12-31",
```

Comment:
Please use any other date string besides `9999-12-31` 😅 

We should avoid adding this specific value to the codebase. If you want to make sure the test expiry_date is always in the future, you can use moment

Reply:
hahaha yes, updated.